### DISCUSSION TITLE
Reference count of a javascript object

### DISCUSSION THREAD

**Discussion by nearest-river** [wasm-bindgen/wasm-bindgen#3973] [Q&A]
### Summary

I need to obtain the reference count of an object..
I'm trying to make a callback function invoke on ref-count change..

### Additional Details

I just need to know where the javascript objects are stored and the memory layout of them..

**Comment by Liamolucko** [comment]
`JsValue`s don't use reference counting. Each `JsValue` contains the index of its value in a global JS array called `heap`: when they're cloned, the JS value gets copied to another spot in the array and the new `JsValue` contains the index of that new spot.

That means that it's left to the JavaScript garbage collector to figure out when there are no more references to a JS value.

**Comment by nearest-river** [comment]
> `JsValue`s don't use reference counting. Each `JsValue` contains the index of its value in a global JS array called `heap`: when they're cloned, the JS value gets copied to another spot in the array and the new `JsValue` contains the index of that new spot.
> 
> That means that it's left to the JavaScript garbage collector to figure out when there are no more references to a JS value.

I know that much..
I just want get where the actual values are stored..
and the memory layout
